### PR TITLE
Add tests for controllers.

### PR DESCRIPTION
**Pull request by TayyabKM**
Following tasks were completed: 

- Add RSpec as a gem to your project.
-  Create a Request spec file for all your controllers.
-  Make sure that for each action you check:
-  If the response status was correct.
-  If a correct template was rendered.
-  If the response body includes correct placeholder text

### CODE REVIEW

File: spec/requests/posts_spec.rb [hunk 1]
```diff
@@ -2,6 +2,19 @@
 
 RSpec.describe 'Posts', type: :request do
   describe 'GET /index' do
-    pending "add some examples (or delete) #{__FILE__}"
+    it 'returns a successful response' do
+      get posts_path
+      expect(response).to have_http_status(200)
+    end
```
Comment (mckent05) [inline spec/requests/posts_spec.rb]: - Good job done so far, but this test is currently failing, please fix this 🙏. I suggest you use this instead `  expect(response).to have_http_status(:success)`
- Also, according to the project requirements you are required to test for all actions in the controller, I suggest you also test for the `post#show` action. Please fix this 🙏

File: spec/requests/users_spec.rb [hunk 2]
```diff
@@ -2,6 +2,19 @@
...
+      get '/users'
+      expect(response).to render_template(:index)
+    end
+
+    it 'includes correct placeholder text in the response body' do
+      get '/users'
+      expect(response.body).to include('Placeholder text for all users')
+    end
   end
 end
```
Comment (mckent05) [inline spec/requests/users_spec.rb]: - Also, according to the project requirements you are required to test for all actions in the controller, I suggest you also test for the `User#show` action. Please fix this 🙏

File: app/controllers/posts_controller.rb [hunk 3]
```diff
@@ -1,11 +1,16 @@
 class PostsController < ApplicationController
   # Placeholder action for all posts by a given user
   def index
-    # Your logic here
+    @posts = Post.all
+    @placeholder_text = 'Placeholder text for all posts by a given user'
```
Comment (mckent05) [inline app/controllers/posts_controller.rb:5]: **OPTIONAL** - Good job done so far, it's not good practice to populate your controller with items that are not needed, please try to keep content in your controller as minimal as possible. I suggest you render this in the respective view file 👍 👍

File: app/controllers/users_controller.rb [hunk 4]
```diff
@@ -2,6 +2,7 @@ class UsersController < ApplicationController
   # Placeholder action for all users page
   def index
     @users = User.distinct
+    @placeholder_text = 'Placeholder text for all users'
```
Comment (mckent05) [inline app/controllers/users_controller.rb:5]: **OPTIONAL** - Good job done so far, it's not good practice to populate your controller with items that are not needed, please try to keep content in your controller as minimal as possible. I suggest you render this in the respective view file 👍 👍

**Review by mckent05 — CHANGES_REQUESTED** [review]
Hi @TayyabKM, 

# STATUS: CHANGES REQUIRED ♻️ ♻️ 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
almost there

## Highlights
- All linter checks are passing ✔️ ✔️ 
- Your PR is professional 👍 👍 
- Good job adding testing your controllers 👍 👍 


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by franclobo — APPROVED** [review]
Hi @TayyabKM, 

# STATUS: APPROVED 😃 

git-merge

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights

- No linter errors ✔️ 
- Descriptive PR ✔️ 
- Correct Git flow ✔️ 
- All tests passed ✔️ 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**